### PR TITLE
sched/group: Simplify the allocation and deallocation logic

### PR DESCRIPTION
**Pull request by xiaoxiang781216**
## Summary
1.Move tg_membe allocation to group_alloc
2.Merge group_deallocate to group_release

## Impact
Code refactor only

## Testing
Pass CI and local test

### CODE REVIEW

**Comment by masayuki2009** [comment]
@xiaoxiang781216 

Hmm, sabre-6quad:knsh has an issue.

```
$ qemu-system-arm -semihosting -M sabrelite -m 1024 -smp 4 -kernel ./nuttx -nographic

ABDGHIJKNOPQ

NuttShell (NSH) NuttX-3.6.1
nsh> ps
  PID GROUP PRI POLICY   TYPE    NPX STATE    EVENT     SIGMASK   STACK   USED  FILLED COMMAND
    0     0   0 FIFO     Kthread N-- Ready              00000000 002024 001000  49.4%  Idle Task
    1     1 100 RR       Task    --- Running            00000000 002016 001232  61.1%  /system/bin/init
nsh> free
                   total       used       free    largest  nused  nfree
        Umem:    1048224       3728    1044496    1038304      6      2
        Kmem: 1065201200      10720 1065190480 1065190480     30      1
        Page:  134217728    1122304  133095424  133095424
sh> /system/bin/hello
Hello, World!!
[   21.590000] arm_dataabort: Data abort. PC: 1080655c DFAR: 80100004 DFSR: 00000005
[   21.590000] up_assert: Assertion failed at file:armv7-a/arm_dataabort.c line: 161 task: /system/bin/hello
[   21.590000] arm_registerdump: R0: 80202000 R1: 00000000 R2: 00000000  R3: 80100000
[   21.590000] arm_registerdump: R4: 00000000 R5: 00000000 R6: 00000000  FP: 108289c0
[   21.590000] arm_registerdump: R8: 00000000 SB: 00000000 SL: 00000000 R11: 00000000
[   21.590000] arm_registerdump: IP: 10804d05 SP: 108289c0 LR: 1081f59d  PC: 1080655c
[   21.590000] arm_registerdump: CPSR: 400000f3
[   21.590000] arm_dump_stack: IRQ Stack:
[   21.590000] arm_dump_stack: sp:     10828888
[   21.590000] arm_dump_stack:   base: 10822d80
[   21.590000] arm_dump_stack:   size: 00000800
[   21.590000] arm_dump_stack: ERROR: IRQ Stack pointer is not within the stack
[   21.590000] arm_stackdump: 10822d80: deadbeef deadbeef deadbeef deadbeef deadbeef deadbeef deadbeef deadbeef
```

**Comment by xiaoxiang781216** [comment]
@masayuki2009 I will take a look.

File: sched/group/group_leave.c [hunk 1]
```diff
@@ -241,6 +233,28 @@ static inline void group_release(FAR struct task_group_s *group)
     }
 #endif
 
+#ifdef CONFIG_ARCH_ADDRENV
+  save_addrenv_t oldenv;
```
Comment (xiaoxiang781216) [inline sched/group/group_leave.c:237]: move to the begin of function

File: sched/group/group_leave.c [hunk 2]
```diff
@@ -241,6 +233,28 @@ static inline void group_release(FAR struct task_group_s *group)
...
+
+  /* NOTE: switch the addrenv before accessing group->tg_info
+   * located in the userland, also save the current addrenv
```
Comment (xiaoxiang781216) [inline sched/group/group_leave.c:240]: change to /* Switch the addrenv and also save the current addrenv */

File: sched/group/group_leave.c (hunk 1 above)
Comment (anchao) [reply sched/group/group_leave.c]: Done

**Comment by masayuki2009** [comment]
@xiaoxiang781216  @anchao 

I confirmed the issue is fixed.

**Pull request merged**